### PR TITLE
Implement "dcos plugin list"

### PR DESCRIPTION
**Pull request by bamarni**
It supports table and JSON outputs.

TABLE:

    $ dcos plugin list
             NAME                 COMMANDS
      dcos-enterprise-cli  backup license security

JSON:
```` json
  $ dcos plugin list --json
  [
      {
          "name": "dcos-enterprise-cli",
          "commands": [
              {
                  "name": "backup",
                  "path": "/home/bamarni/.dcos/clusters/a95bf067-2e6b-4468-ba86-0d177b226950/subcommands/dcos-enterprise-cli/env/bin/dcos-backup",
                  "description": "Access DC/OS backup functionality"
              },
              {
                  "name": "license",
                  "path": "/home/bamarni/.dcos/clusters/a95bf067-2e6b-4468-ba86-0d177b226950/subcommands/dcos-enterprise-cli/env/bin/dcos-license",
                  "description": "Manage your DC/OS licenses"
              },
              {
                  "name": "security",
                  "path": "/home/bamarni/.dcos/clusters/a95bf067-2e6b-4468-ba86-0d177b226950/subcommands/dcos-enterprise-cli/env/bin/dcos-security",
                  "description": "DC/OS security related commands"
              }
          ]
      }
  ]
````

https://jira.mesosphere.com/browse/DCOS_OSS-3853

### CODE REVIEW

File: pkg/cmd/dcos_test.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ Commands:
       Manage the DC/OS configuration file
   help
       Help about any command
+  plugin
+      Manage CLI plugins
```
Comment (ArmandGrillet) [inline pkg/cmd/dcos_test.go]: Should this be in a separate PR/commit?

File: pkg/plugin/plugin.go [hunk 2]
```diff
@@ -2,13 +2,13 @@ package plugin
 
 // Plugin defines an external plugin and its associated data.
```
Comment (ArmandGrillet) [inline pkg/plugin/plugin.go]: Adding that we have toml keys for input and json keys for output could be useful.

File: pkg/cmd/dcos_test.go (hunk 1 above)
Comment (bamarni) [reply pkg/cmd/dcos_test.go]: This was present in different PRs, as I've merged the `dcos plugin remove` PR it is not in this one anymore.

File: pkg/plugin/plugin.go (hunk 2 above)
Comment (bamarni) [reply pkg/plugin/plugin.go]: I've added some notes about it.
